src/plugins/visualizations/public/index.ts

```diff
@@ -49,3 +49,12 @@ export {
   FakeParams,
   HistogramParams,
 } from '../common/expression_functions/xy_dimension';
+export {
+  convertToSerializedVis,
```

Comment:
It seems some of these are already exposed via plugin contract instead of exported as static function - I think for these a plugin contract makes more sense.

Could you please restructure this as follows:
* Remove top level exports
* Add missing pieces to the `start` contract of the visualizations plugin
* Spread in arguments which are simply contracts of other plugins in the visualizations plugin class instead of requiring them from the caller:
 
Instead of 
```ts
  const savedVis: VisSavedObject = await getSavedVisualization(
    {
      search: data.search,
      savedObjectsClient: savedObjects.client,
      dataViews: data.dataViews,
      spaces,
      savedObjectsTagging,
    },
    opts
  );
```

the caller should do

```ts
  const savedVis: VisSavedObject = await visualizations.getSavedVisualization(opts);
```

in the visualizations plugin itself it can stay a static function, it should just be called like this in the plugin class:
```ts
      getSavedVisualization: async (opts) =>
        await getSavedVisualization({
      search: data.search,
      savedObjectsClient: savedObjects.client,
      dataViews: data.dataViews,
      spaces,
      savedObjectsTagging,
    }, opts),
```

Reply:
Done